marketplace/deployer_helm_tiller_base/bin/deploy_internal.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eox pipefail
+
+app_uid=$(kubectl get "applications/$NAME" \
+  --namespace="$NAMESPACE" \
+  --output=jsonpath='{.metadata.uid}')
+app_api_version=$(kubectl get "applications/$NAME" \
+  --namespace="$NAMESPACE" \
+  --output=jsonpath='{.apiVersion}')
+
+for chart in /data/chart/*; do
+  # Expand and apply the template for the Application resource.
+  # Note: This must be done out-of-band because the Application resource
+  # is created before the deployer is invoked, but helm does not handle
+  # pre-existing resources.
+  helm template \
+      --name="$NAME" \
+      --namespace="$NAMESPACE" \
+      --values=<(print_config.py --output=yaml) \
+      --set "$(cat /data/schema.yaml \
+                 | yaml2json \
+                 | jq -r '.properties
+                            | to_entries
+                            | .[]
+                            | select(.value["x-google-marketplace"].type == "APPLICATION_UID")
+                            | .key')=" \
+      "$chart" \
+    | yaml2json \
+    | jq 'select( .kind == "Application" )' \
+    | kubectl apply -f -
+
+  # Install the chart.
+  # Note: This does not assume that tiller is installed in the cluster;
+  # it runs a local tiller process to perform template expansion and
+  # hook processing.
+  # Note: The local tiller process is configured with --storage=secret,
+  # which is recommended but not default behavior.
+  helm tiller run "$NAMESPACE" -- \
+    helm install \
+        --name="$NAME" \
+        --namespace="$NAMESPACE" \
+        --values=<(print_config.py --output=yaml) \
+        "$chart"
+
+  # Establish an ownerReference back to the Application resource, so that
+  # the helm release will be cleaned up when the Application is deleted.
+  # Note: This is fragile, as any time that tiller updates the release
```

Comment:
hmm, shouldn't this then be in the post install hook? It should be able to query the application object using labels.

Reply:
Unfortunately it can't - touching the secret from within the hook causes a failure.